autoload/go/job.vim

```diff
@@ -57,16 +57,17 @@ function go#job#Spawn(args)
       endif
     endif
 
+    let l:listtype = "quickfix"
```

Comment:
I think we should use here:  `let l:listtype = go#list#Type("quickfix")`. the `Type()` function returns the user preference (can be set via `go_list_type`) , if there is none, then the argument itself is returned.